runtime/vm/resolvesupport.cpp

```diff
@@ -1851,6 +1851,131 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveConstantDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	J9RAMConstantDynamicRef *ramCPEntry = (J9RAMConstantDynamicRef*)ramCP + cpIndex;
+	j9object_t value = NULL;
+	J9JavaVM *vm = vmThread->javaVM;
+	bool resolved = false;
+
+RETRY:
+	/* Fast path resolution for previously resolved entries */
+	if (NULL != ramCPEntry->value) {
```

Comment:
This is the expected case, so could read into value, and NULL check the local.  Also means value could be declared and initialized directly here.